### PR TITLE
add support to list volumes based on metadata

### PR DESCRIPTION
**Pull request by Madhu-1**
* if volume name is not present , list all volumes
* if name is present get that volume info
* if key, value are present, list volumes
based on key and value

Signed-off-by: Madhu Rajanna <mrajanna@redhat.com>

### CODE REVIEW

**Review by sac-urs — CHANGES_REQUESTED** [review]

File: glusterapi/volume.py [hunk 1]
```diff
@@ -58,6 +58,7 @@ def volume_create(self, volume_name="", bricks=None,
         :param arbiter  (int) arbiter count
         :param force  (bool)  force flag
         :param options (dict) volume options
+        :param metadata (dict) volume metadata
         :raises: GlusterApiError or GlusterApiInvalidInputs on failure
         """
         if len(bricks) <= 0:
```
[the remark this answers is not in the document]
Comment (Madhu-1) [reply glusterapi/volume.py]: yeah, Thanks for pointing it out.

**Review by sac-urs — APPROVED** [review]
Looks good to me.

**Review by devyanikota — APPROVED** [review]

**Comment by devyanikota** [comment]
+1

**Comment by Madhu-1** [comment]
@aravindavk please review

**Pull request merged**